presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1153,13 +1154,17 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
             }
         }
 
-        Domain domain = Domain.multipleValues(type, nonNullValues);
+        if (!nonNullValues.isEmpty()) {
+            Domain domain = Domain.multipleValues(type, nonNullValues);
+            if (hasNull) {
+                return domain.union(Domain.onlyNull(type));
+            }
 
-        if (hasNull) {
-            domain = domain.union(Domain.onlyNull(type));
+            return domain;
         }
 
-        return domain;
+        checkState(hasNull, "partition column must contain a null or non-null value");
```

Comment:
The only way this would happen is if `partitions` is empty. Maybe check for that at the beginning of the method, instead.